docker-build.sh

```diff
@@ -50,59 +57,74 @@ fi
 # NOTE: elasticsearch only keeps the newest minor version tag.  Older
 # minor version tags are deleted when a newer minor version is released
 
-
 # can be specific or just a major version number.  As long as there 
 # exists a docker tag for the string you enter
 # ES_VERSIONS="2.4"
 
-# List of plugins to install can be a specific version or just the
-# name and script will just grab newest version.
-
-# ES_PLUGINS="royrusso/elasticsearch-HQ/v2.0.3 cloud-gce mobz/elasticsearch-head "
-# ES_PLUGINS="cloud-gce"
-
 if [ "${FORCE_BUILD}" -ne 0 ]
 then
    echo "FORCING BUILD of all versions"
 fi
 
-#for version in "${ES_VERSIONS}"
 egrep "^[a-zA-Z0-9]+" ${config_file} | while read line
 do
   # decode config line
-  version=`echo $line | cut -d ':' -f1`
-  plugins=`echo $line | cut -d ':' -f2`
+  # first word is version string
+  version=`first $line`
+  # the rest (if there is anymore are space separated plugins)
+  plugins=`rest $line`
+  # default version option
+  vers_opt="-version"
 
   # TODO maybe make config file more forgiving
   #  - support colon separated plugins,
   #  - supprot common separated plugins
   #  - support multiple separators common, colon, space
 
+  # get major version number since each major has significant chnages
+  # that impact build.
+  ver_maj=$(echo $version | cut -d '.' -f1)
+
+  # elasticsearch changed their docker repo from docker hub to their own
+  #  for versions 5.x and newer.  version less than 5 use docker hub
+  case ${ver_maj} in
+    "1") # docker hub and -v
+          DOCKER_REPO=""
+          vers_opt="-v"
+        ;;
+    "2") # use docker hub
+          DOCKER_REPO=""
+        ;;
+  esac
```

Comment:
- Add ES 5 section to case statement (see above review comment).
- Add wildcard section to case statement to catch unsupported values of maj_ver, and exit with a return value > 1.